homeassistant/components/xiaomi_ble/config_flow.py

```diff
@@ -39,25 +40,102 @@ async def async_step_bluetooth(
             return self.async_abort(reason="not_supported")
         self._discovery_info = discovery_info
         self._discovered_device = device
+
+        title = device.title or device.get_device_name() or discovery_info.name
+        self.context["title_placeholders"] = {"name": title}
+
+        if device.encryption_scheme == EncryptionScheme.MIBEACON_LEGACY:
+            return await self.async_step_get_encryption_key_legacy()
+        if device.encryption_scheme == EncryptionScheme.MIBEACON_4_5:
+            return await self.async_step_get_encryption_key_4_5()
         return await self.async_step_bluetooth_confirm()
 
+    async def async_step_get_encryption_key_legacy(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Enter a legacy bindkey for a v2/v3 MiBeacon device."""
+        assert self._discovery_info
+
+        errors = {}
+
+        if user_input is not None:
+            bindkey = user_input["bindkey"]
+
+            if len(bindkey) != 24:
+                errors["bindkey"] = "expected_24_characters"
+            else:
+                device = DeviceData(bindkey=user_input["bindkey"])
+
+                # If we got this far we already know supported will
+                # return true so we don't bother checking that again
+                # We just want to retry the decryption
+                device.supported(self._discovery_info)
+
+                if device.bindkey_verified:
+                    return self.async_create_entry(
+                        title=self.context["title_placeholders"]["name"],
+                        data={"bindkey": user_input["bindkey"]},
+                    )
+
+                errors["bindkey"] = "decryption_failed"
+
+        return self.async_show_form(
+            step_id="get_encryption_key_legacy",
+            description_placeholders=self.context["title_placeholders"],
+            data_schema=vol.Schema({vol.Required("bindkey"): vol.All(str, vol.Strip)}),
```

Comment:
Please update strings.json and add the step form data items for the new keys.

Reply:
I see you've got it on the todo already.

Reply:
Haha, thanks. I've added them now. It's been a while since i've updated strings.json - do i need to update `translations/en.json` or will that happen automatically?

Reply:
Run `python3 -m script.translations develop` to update them.

Reply:
You don't need to update that file, but it's allowed to ease local dev testing. In that case run:

```sh
python -m script.translations develop
```